### PR TITLE
Add tests for Control (read, update, Link header)

### PR DESCRIPTION
**Pull request by ylebre**

### CODE REVIEW

File: test/surface/read-logged_in.test.ts [hunk 1]
```diff
@@ -245,5 +245,50 @@ describe('Read-LoggedIn', () => {
...
+      body: makeBody('acl:Read', 'acl:Read', resourceUrl),
+      headers: {
+        'Content-Type': 'text/turtle',
+        // 'If-None-Match': '*' - work around a bug in some servers that don't support If-None-Match on ACL doc URLs
+      }
+    });
+    const result = await solidLogicBob.fetch(aclDocUrl);
+    expect(responseCodeGroup(result.status)).toEqual("4xx");
+  });
+  it('Is allowed to read .acl with Control+Read', async () => {
```
Comment (michielbdejong) [inline test/surface/read-logged_in.test.ts:271]: It should even be allowed with just Control. Read only applies to reading the resource itself.

Control access is everything (create, read, update, delete) or nothing for just the ACL doc associated with the resource.

File: test/surface/read-logged_in.test.ts [hunk 2]
```diff
@@ -245,5 +245,50 @@ describe('Read-LoggedIn', () => {
...
+        'If-None-Match': '*'
+      }
+    });
+    const aclDocUrl = await solidLogicAlice.findAclDocUrl(resourceUrl);
+    await solidLogicAlice.fetch(aclDocUrl, {
+      method: 'PUT',
+      body: makeBody('acl:Control, acl:Read', 'acl:Control, acl:Read', resourceUrl),
+      headers: {
+        'Content-Type': 'text/turtle',
+        // 'If-None-Match': '*' - work around a bug in some servers that don't support If-None-Match on ACL doc URLs
```
Comment (michielbdejong) [inline test/surface/read-logged_in.test.ts:288]: I'm thinking maybe we shouldn't call this a "bug". We can say "The URL of the ACL doc we are creating may differ only in the query part from a resource that already exists, so we can't expect If-None-Match to work for ACLs".

@acoburn what do you think?

File: test/surface/read.test.ts [hunk 3]
```diff
@@ -247,5 +247,50 @@ describe('Read', () => {
...
+      body: makeBody('acl:Read', 'acl:Read', resourceUrl),
+      headers: {
+        'Content-Type': 'text/turtle',
+        // 'If-None-Match': '*' - work around a bug in some servers that don't support If-None-Match on ACL doc URLs
+      }
+    });
+    const result = await solidLogicBob.fetch(aclDocUrl);
+    expect(responseCodeGroup(result.status)).toEqual("4xx");
+  });
+  it('Is allowed to read .acl with Control+Read', async () => {
```
Comment (michielbdejong) [inline test/surface/read.test.ts:273]: Same as above, change from 'Control+Read' to 'Control'.

File: test/surface/update.test.ts [hunk 4]
```diff
@@ -771,6 +771,69 @@ describe('Update', () => {
...
+        method: 'PUT',
+        body: makeBody(null, 'acl:Read', containerUrl),
+        headers: {
+          'Content-Type': 'text/turtle',
+          // 'If-None-Match': '*' - work around a bug in some servers that don't support If-None-Match on ACL doc URLs
+        }
+      });
+      expect(result.status).toEqual(403);
+    });
+    it('Is allowed with Read+Control+Write', async () => {
```
Comment (michielbdejong) [inline test/surface/update.test.ts:807]: Same as above, change from 'Read+Control+Write' to 'Control'.

File: test/surface/wac-allow.test.ts [hunk 5]
```diff
@@ -192,5 +192,15 @@ describe('From default', () => {
       const result = await fetch(`${testFolderUrl}3/publicReadBobWrite/test.txt`);
       expect(sortWac(result.headers.get('WAC-Allow'))).toEqual(sortWac('user="read append",public="read append"'));
     });
+    it(`Shows the Link header containing the aclDocUrl to Alice`, async () => {
+      const result = await solidLogicBob.fetch(`${testFolderUrl}3/publicReadBobWrite/test.txt`);
+      const aclDocUrl = await solidLogicAlice.findAclDocUrl(`${testFolderUrl}3/publicReadBobWrite/test.txt`);
+      expect(result.headers.get('Link')).toContain(aclDocUrl);
+    });
+    it(`Does not show a Link header containing the aclDocUrl to the public`, async () => {
```
Comment (michielbdejong) [inline test/surface/wac-allow.test.ts:200]: OK to add the code for this test but please add it with `it.skip` and add a comment 'DISPUTED: See https://github.com/solid/specification/pull/248

**Review by michielbdejong — COMMENTED** [review]
Awesome work @ylebre!
This was really needed and moves us forward a great deal.

LGTM, other than two issues:

* only give `acl:Control` and nothing else in tests that should succeed
* skip the test about public discovery

File: test/surface/wac-allow.test.ts (hunk 5 above)
Comment (ylebre) [reply test/surface/wac-allow.test.ts:200]: Marking these as skipped with the comment and link to the dispute.

File: test/surface/read-logged_in.test.ts (hunk 1 above)
Comment (ylebre) [reply test/surface/read-logged_in.test.ts:271]: Right, changing this to Control only.

File: test/surface/read-logged_in.test.ts (hunk 2 above)
Comment (ylebre) [reply test/surface/read-logged_in.test.ts:288]: This comment came from the test I used as a template, removing this from here;

File: test/surface/read.test.ts (hunk 3 above)
Comment (ylebre) [reply test/surface/read.test.ts:273]: Changing to Control only.

File: test/surface/read-public.test.ts [hunk 6]
```diff
@@ -247,5 +247,50 @@ describe('Read-Public', () => {
...
+      body: makeBody('acl:Read', 'acl:Read', resourceUrl),
+      headers: {
+        'Content-Type': 'text/turtle',
+        // 'If-None-Match': '*' - work around a bug in some servers that don't support If-None-Match on ACL doc URLs
+      }
+    });
+    const result = await solidLogicBob.fetch(aclDocUrl);
+    expect(responseCodeGroup(result.status)).toEqual("4xx");
+  });
+  it('Is allowed to read .acl with Control+Read', async () => {
```
[the remark this answers is not in the document]
Comment (ylebre) [reply test/surface/read-public.test.ts]: Updated to control only.

File: test/surface/update.test.ts (hunk 4 above)
Comment (ylebre) [reply test/surface/update.test.ts]: Updated to Control only

File: test/surface/wac-allow.test.ts [hunk 7]
```diff
@@ -192,12 +192,15 @@ describe('From default', () => {
       const result = await fetch(`${testFolderUrl}3/publicReadBobWrite/test.txt`);
       expect(sortWac(result.headers.get('WAC-Allow'))).toEqual(sortWac('user="read append",public="read append"'));
     });
-    it(`Shows the Link header containing the aclDocUrl to Alice`, async () => {
+
+    // DISPUTED: See https://github.com/solid/specification/pull/248
+    it.skip(`Shows the Link header containing the aclDocUrl to Alice`, async () => {
```
Comment (michielbdejong) [inline test/surface/wac-allow.test.ts:197]: This one is not disputed, only the other one.

**Pull request merged**